src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtil.java

```diff
@@ -380,12 +380,27 @@ public static boolean isNonVoidMethod(DetailAST methodDefAst) {
     /**
      * Checks whether a parameter is a receiver.
      *
+     * <p>A receiver parameter is a special parameter that
+     * represents the object for which the method is invoked.
+     * It is denoted by the reserved keyword {@code this}
+     * in the method declaration.
+     * For example, the following method has a receiver parameter:
+     *
+     * <pre>
+     * public class Param {
+     *   public void foo(Param this) {
```

Comment:
instead of this code block, please addlink to javadoc of TokenTypes.java#PARAMETER_DEF
it already has all examples of java code and ast.

Reply:
done